### PR TITLE
Allow specification of Kubernetes plugin version

### LINKED ISSUE #513: Difficult to select version of Kubernetes plugin to use
Currently, we can install our Kubernetes integration using the `--kubernetes` option on `calicoctl node`.

However, the calico-kubernetes plugin version is hard-coded in calicoctl with no way to configure it.  This is inflexible and leads to difficulty upgrading the plugin, which makes it hard to deploy bugfixes to users (a new release of calicoctl is required)

A few options here:
- `calicoctl node --kubernetes=version`
  - problem: docopt would require a value for `version`, so we couldn't set a default of `latest`.
- `calicoctl node --kubernetes --plugin-version=version` or something like that
- We could eliminate the `--kubernetes` flag altogether, use @Symmetric's work to containerize the kubernetes plugin and use DockerHub for distribution.
- ... ???

### PR DESCRIPTION
**Pull request by caseydavenport**
Fixes #513

### CODE REVIEW

**Comment by caseydavenport** [comment]
Fixes #513

File: calico_containers/calico_ctl/node.py [hunk 1]
```diff
@@ -191,12 +195,13 @@ def node(arguments):
...
                    detach=detach,
                    kubernetes=arguments.get("--kubernetes"),
+                   kubernetes_version=kubernetes_version,
                    rkt=arguments.get("--rkt"),
                    libnetwork_image=libnetwork_image)
 
 
-def node_start(node_image, log_dir, ip, ip6, as_num, detach, kubernetes, rkt,
-               libnetwork_image):
+def node_start(node_image, log_dir, ip, ip6, as_num, detach, kubernetes,
```
Comment (robbrockbank) [inline calico_containers/calico_ctl/node.py]: Don't need separate kubernetes boolean as version will be None if kubernetes is False.

File: calico_containers/calico_ctl/node.py [hunk 2]
```diff
@@ -191,12 +195,13 @@ def node(arguments):
...
+               kubernetes_version, rkt, libnetwork_image):
     """
     Create the calico-node container and establish Calico networking on this
```
Comment (robbrockbank) [inline calico_containers/calico_ctl/node.py]: Doc string needs updating.

File: calico_containers/calico_ctl/node.py [hunk 3]
```diff
@@ -636,3 +646,23 @@ def install_plugin(plugin_dir, binary_url):
         except OSError:
             print "ERROR: Unable to set permissions on plugin %s" % binary_path
             sys.exit(1)
+
+
+class URLGetter(urllib.FancyURLopener):
```
Comment (robbrockbank) [inline calico_containers/calico_ctl/node.py]: worthy of putting in utils i think

File: calico_containers/calico_ctl/node.py [hunk 4]
```diff
@@ -636,3 +646,23 @@ def install_plugin(plugin_dir, binary_url):
...
+    """
+    Used to retrieve Kubernetes plugin binaries from GitHub.  Overridden
```
Comment (robbrockbank) [inline calico_containers/calico_ctl/node.py]: its a little more generic than described.

**Comment by robbrockbank** [comment]
minor comments.

File: calico_containers/calico_ctl/node.py (hunk 1 above)
Comment (caseydavenport) [inline calico_containers/calico_ctl/node.py]: Done.

File: calico_containers/calico_ctl/node.py (hunk 2 above)
Comment (caseydavenport) [inline calico_containers/calico_ctl/node.py]: Done.

File: calico_containers/calico_ctl/node.py (hunk 3 above)
Comment (caseydavenport) [inline calico_containers/calico_ctl/node.py]: Done.

File: calico_containers/calico_ctl/node.py (hunk 4 above)
Comment (caseydavenport) [inline calico_containers/calico_ctl/node.py]: Fixed.

**Comment by caseydavenport** [comment]
Should be all fixed up and ready for another go.

File: calico_containers/calico_ctl/utils.py [hunk 5]
```diff
@@ -209,3 +210,23 @@ def convert_asn_to_asplain(asn):
         asn = 65536*int(left_asn)+int(right_asn)
 
     return asn
+
+
+class URLGetter(urllib.FancyURLopener):
+    """
+    Retrieve plugin binaries from GitHub.  Overridden
```
Comment (robbrockbank) [inline calico_containers/calico_ctl/utils.py]: so close  ;-)...... but I don;t think the GitHub comment is relevant is it... ?

**Comment by robbrockbank** [comment]
one teeny comment, but otherwise LGTM

File: calico_containers/calico_ctl/utils.py (hunk 5 above)
Comment (caseydavenport) [inline calico_containers/calico_ctl/utils.py]: Aw shoot.

**Pull request merged**